x-pack/plugins/ingest_manager/server/services/agents/checkin/state_new_actions.ts

```diff
@@ -104,15 +110,29 @@ async function getOrCreateAgentDefaultOutputAPIKey(
   return outputAPIKey.key;
 }
 
-async function createAgentActionFromPolicyAction(
+export async function createAgentActionFromPolicyAction(
   soClient: SavedObjectsClientContract,
   agent: Agent,
   policyAction: AgentPolicyAction
 ) {
+  // Transform the policy action for agent version <=  7.9 for BWC
+  const agentVersion = semver.parse((agent.local_metadata?.elastic as any)?.agent?.version);
+  const agentPolicyAction: AgentPolicyAction | AgentPolicyActionV7_9 =
+    agentVersion && semver.lt(agentVersion, '7.10.0')
```

Comment:
Should we already have an issue to keep of track of these kind of change that need to be removed later?

Reply:
I will start a meta issue